service/matching/matching_engine.go

```diff
@@ -226,11 +235,67 @@ func (e *matchingEngineImpl) Stop() {
 		return
 	}
 
+	e.serviceResolver.RemoveListener(e.listenerKey())
```

Comment:
```suggestion
	_ = e.serviceResolver.RemoveListener(e.listenerKey())
```

Same here. We don't care about the error so let's explicitly show that in the code